indico/MaKaC/PDFinterface/conference.py

```diff
@@ -643,715 +632,742 @@ def _processTOCPage(self):
+        style.fontSize = modifiedFontSize(fSize, self._fontsize)
+        style.leading = modifiedFontSize(fSize + 3, self._fontsize)
+        return Paragraph(text, style)
 
     def _fontifyRow(self, row, fSize=10, fName=""):
-        newrow = []
-        for text in row:
-            newrow.append(self._fontify(text,fSize,fName))
-        return newrow
-
-    def _processDayEntries(self,day,story):
-        res=[]
-        originalts=TableStyle([('VALIGN',(0,0),(-1,-1),"TOP"),
-                        ('LEFTPADDING',(0,0),(-1,-1),1),
-                        ('RIGHTPADDING',(0,0),(-1,-1),1),
-                        ('GRID',(0,1),(-1,-1),1,colors.lightgrey)])
-        self._tsSpk=TableStyle([("LEFTPADDING",(0,0),(0,-1),0),
-                            ("RIGHTPADDING",(0,0),(0,-1),0),
-                            ("TOPPADDING",(0,0),(0,-1),1),
-                            ("BOTTOMPADDING",(0,0),(0,-1),0)])
-        colorts=TableStyle([("LEFTPADDING",(0,0),(-1,-1),3),
-                            ("RIGHTPADDING",(0,0),(-1,-1),0),
-                            ("TOPPADDING",(0,0),(-1,-1),0),
-                            ("BOTTOMPADDING",(0,0),(-1,-1),0),
-                            ('GRID',(0,0),(0,-1),1,colors.lightgrey)])
-        entriesOnDay=self._conf.getSchedule().getEntriesOnDay(day)
-        entriesOnDay.sort(sortEntries)
-        for entry in entriesOnDay:
-            #Session slot
-            if isinstance(entry,schedule.LinkedTimeSchEntry) and \
-                    isinstance(entry.getOwner(),conference.SessionSlot):
-                sessionSlot=entry.getOwner()
-                if len(self._showSessions)>0 and \
-                    sessionSlot.getSession().getId() not in self._showSessions:
+        return [self._fontify(text, fSize, fName) for text in row]
+
+    def _processDayEntries(self, day, story):
+        res = []
+        originalts = TableStyle([('VALIGN', (0, 0), (-1, -1), "TOP"),
+                                 ('LEFTPADDING', (0, 0), (-1, -1), 1),
+                                 ('RIGHTPADDING', (0, 0), (-1, -1), 1),
+                                 ('GRID', (0, 1), (-1, -1), 1, colors.lightgrey)])
+        self._tsSpk = TableStyle([("LEFTPADDING", (0, 0), (0, -1), 0),
+                                  ("RIGHTPADDING", (0, 0), (0, -1), 0),
+                                  ("TOPPADDING", (0, 0), (0, -1), 1),
+                                  ("BOTTOMPADDING", (0, 0), (0, -1), 0)])
+        colorts = TableStyle([("LEFTPADDING", (0, 0), (-1, -1), 3),
+                              ("RIGHTPADDING", (0, 0), (-1, -1), 0),
+                              ("TOPPADDING", (0, 0), (-1, -1), 0),
+                              ("BOTTOMPADDING", (0, 0), (-1, -1), 0),
+                              ('GRID', (0, 0), (0, -1), 1, colors.lightgrey)])
+
+        entries = (self._conf.as_event.timetable_entries
+                   .filter(db.func.cast(TimetableEntry.start_dt, db.Date) == day.date(),
```

Comment:
This is not timezone-friendly. `start_dt` will most probably be in UTC, which may be several hours off from the timezone of `day`. That may result in getting things on the wrong day. You can check the code of the meeting timetable themes, but the idea is to use something like `Event.starts_between`, only applied to entries (you can maybe use `db_dates_overlap` directly). /cc @ThiefMaster, who wrote `starts_between`.